sections/shared/Layout/AppLayout/Header/BalanceActions.tsx

```diff
@@ -1,250 +1,275 @@
-			.filter((position) => position.remainingMargin.gt(zeroBN))
-			.map((position) => setMarketConfig(position, FuturesAccountTypes.CROSS_MARGIN));
-		return [
-			{
-				label: 'header.balance.total-margin-label',
-				totalAvailableMargin: formatDollars(portfolio.total),
-				options: [...isolatedPositionsFiltered, ...crossPositionsFiltered],
-			},
-		];
-	}, [crossPositions, isolatedPositions, setMarketConfig, portfolio]);
-
-	const OptionsGroupLabel: FC<{ label: string; totalAvailableMargin?: string }> = ({
-		label,
-		totalAvailableMargin,
-	}) => (
-		<FlexDivRow>
-			<Container>{t(label)}</Container>
-			<Container>{totalAvailableMargin}</Container>
-		</FlexDivRow>
-	);
+	const claimDisabledKwentaOp = useMemo(() => kwentaOpRewards.lte(0), [kwentaOpRewards]);
 
-	const formatOptionLabel: FC<ReactSelectOptionProps> = ({
-		label,
-		synthIcon,
-		marketRemainingMargin,
-		onClick,
-	}: ReactSelectOptionProps) => (
-		<LabelContainer onClick={onClick}>
-			<FlexDivRow>
-				{synthIcon && <StyledCurrencyIcon currencyKey={synthIcon} width="24px" height="24px" />}
-				<StyledLabel>{t(label)}</StyledLabel>
-			</FlexDivRow>
-			<Container>{marketRemainingMargin}</Container>
-		</LabelContainer>
-	);
+	const claimDisabledSnxOp = useMemo(() => snxOpRewards.lte(0), [snxOpRewards]);
+
+	const REWARDS = [
+		{
+			key: 'trading-rewards',
+			title: t('dashboard.rewards.trading-rewards.title'),
+			copy: t('dashboard.rewards.trading-rewards.copy'),
+			button: t('dashboard.rewards.staking'),
+			kwentaIcon: true,
+			linkIcon: true,
+			rewards: tradingRewards,
+			onClick: goToStaking,
+			isDisabled: false,
+		},
+		{
+			key: 'kwenta-rewards',
+			title: t('dashboard.rewards.kwenta-rewards.title'),
+			copy: t('dashboard.rewards.kwenta-rewards.copy'),
+			button: t('dashboard.rewards.claim'),
+			kwentaIcon: false,
+			linkIcon: false,
+			rewards: kwentaOpRewards,
+			onClick: handleClaimOp,
+			isDisabled: claimDisabledKwentaOp,
+		},
+		{
+			key: 'snx-rewards',
+			title: t('dashboard.rewards.snx-rewards.title'),
+			copy: t('dashboard.rewards.snx-rewards.copy'),
+			button: t('dashboard.rewards.claim'),
+			kwentaIcon: false,
+			linkIcon: false,
+			rewards: snxOpRewards,
+			onClick: () => {},
+			isDisabled: claimDisabledSnxOp,
+		},
+	];
 
 	useEffect(() => {
-		setBalanceLabel(formatDollars(susdWalletBalance, { sign: '$' }));
-	}, [balanceLabel, susdWalletBalance]);
+		const tokenAddresses = [KWENTA_ADDRESS, OP_ADDRESS];
+		const initExchangeTokens = async () => {
+			const coinGeckoPrices = await sdk.exchange.batchGetCoingeckoPrices(tokenAddresses, true);
+			const [kwentaPrice, opPrice] = tokenAddresses.map(
+				(tokenAddress) => coinGeckoPrices[tokenAddress]?.usd.toString() ?? 0
+			);
 
-	if (!balanceLabel) {
-		return null;
-	}
+			setRewardBalance(
+				toWei(kwentaPrice)
+					.mul(tradingRewards)
+					.add(toWei(opPrice).mul(kwentaOpRewards.add(snxOpRewards)))
+			);
+		};
+
+		(async () => {
```

Comment:
Let's try and always avoid anon functions but this relates to the comment above anyway so we can remove altogether.